File: app/src/main/java/com/chaitupenju/popularmovies2/reviews/Review.java
package com.chaitupenju.popularmovies2.reviews;

import android.util.Log;

import java.util.ArrayList;

public class Review {
     String author;
     String content;

    private static final String delimitStr = "-TS-";
    private static final String delimitReviewStr = "-ROBJ-";
    public Review(String author, String content) {
        this.author = author;
        this.content = content;
    }

    /*
    my string will be like this
    author0,content0-RS-author1,content1-RS-author2,content2
    */
    public static ArrayList<Review> stringToArray(String reviewStr){
        ArrayList<Review> retReviewArr = new ArrayList<>();
        String revParts[] = reviewStr.split(delimitStr);
        for(String review: revParts){
            String revs[] = review.split(delimitReviewStr);
            try{
                retReviewArr.add(new Review(revs[0], revs[1]));
            }catch (IndexOutOfBoundsException e){
                e.printStackTrace();
            }
        }
        Log.i("STR2ARR",retReviewArr.toString());
        return retReviewArr;
    }

    public static String arrayToString(ArrayList<Review> reviewArr){
        StringBuilder retStr = new StringBuilder();
        try{
            for (int i = 0; i < reviewArr.size(); i++) {
                retStr.append(reviewArr.get(i).author).append(delimitReviewStr).append(reviewArr.get(i).content);
                if (i < reviewArr.size() - 1) {
                    retStr.append(delimitStr);
                }
            }
        }
        catch(NullPointerException npe){
            return "";
        }
        return retStr.toString();
    }
}

File: app/src/main/java/com/chaitupenju/popularmovies2/models/Review.kt
package com.chaitupenju.popularmovies2.models

import android.util.Log


data class Review(
    val author: String,
    val content: String
) {

    companion object {
        private const val delimitStr = "-TS-"
        private const val delimitReviewStr = "-ROBJ-"

        fun stringToArray(reviewStr: String): ArrayList<Review> {
            val retReviewArr = ArrayList<Review>()
            val revParts: Array<String> = reviewStr.split(delimitStr).toTypedArray()
            for (review in revParts) {
                val revs: Array<String> = review.split(delimitReviewStr).toTypedArray()
                try {
                    retReviewArr.add(Review(revs[0], revs[1]))
                } catch (e: IndexOutOfBoundsException) {
                    e.printStackTrace()
                }
            }
            Log.i("STR2ARR", retReviewArr.toString())
            return retReviewArr
        }

        fun arrayToString(reviewArr: java.util.ArrayList<Review>): String {
            val retStr = StringBuilder()
            try {
                for (i in reviewArr.indices) {
                    retStr.append(reviewArr[i].author).append(delimitReviewStr).append(
                        reviewArr[i].content
                    )
                    if (i < reviewArr.size - 1) {
                        retStr.append(delimitStr)
                    }
                }
            } catch (npe: NullPointerException) {
                return ""
            }
            return retStr.toString()
        }
    }
}